src/v2/Apps/Auction/Components/RegistrationForm.tsx

```diff
@@ -1,124 +1,91 @@
-import { Box, Button, Flex, Sans, Serif, Spacer } from "@artsy/palette"
-import { Form, Formik, FormikConfig } from "formik"
+import { Box, Button, Flex, Text, Spacer } from "@artsy/palette"
+import { Form, FormikConfig, FormikProps } from "formik"
 import * as React from "react"
 
 import { CreditCardInstructions } from "v2/Apps/Auction/Components/CreditCardInstructions"
 import { CreditCardInput } from "v2/Apps/Order/Components/CreditCardInput"
-import {
-  AddressErrors,
-  AddressForm,
-  AddressTouched,
-} from "v2/Components/AddressForm"
+import { AddressForm } from "v2/Components/AddressForm"
 import { ConditionsOfSaleCheckbox } from "v2/Components/Auction/ConditionsOfSaleCheckbox"
-import { ErrorModal } from "v2/Components/Modal/ErrorModal"
-import {
-  FormValuesForRegistration,
-  Registration,
-  initialValuesForRegistration,
-} from "v2/Apps/Auction/Components/Form"
+import { BillingInfoFormValues, BillingInfoWithTerms } from "./Form"
 import {
   OnSubmitValidationError,
   TrackErrors,
 } from "v2/Apps/Auction/Components/OnSubmitValidationError"
-import type { StripeError } from "@stripe/stripe-js"
+import { BillingInfoFormContext } from "./Form"
+import { ErrorModal } from "v2/Components/Modal/ErrorModal"
 
+type FormValues = Pick<
+  BillingInfoFormValues,
+  "address" | "agreeToTerms" | "creditCard"
+>
 interface RegistrationFormProps {
-  onSubmit: FormikConfig<FormValuesForRegistration>["onSubmit"]
+  onSubmit: FormikConfig<FormValues>["onSubmit"]
   trackSubmissionErrors: TrackErrors
   needsIdentityVerification: boolean
 }
+export const RegistrationForm: React.FC<RegistrationFormProps> = ({
+  onSubmit,
+  trackSubmissionErrors,
+  needsIdentityVerification,
+}) => {
+  return (
+    <BillingInfoFormContext
+      formKeys={["addressWithPhone", "agreeToTerms", "creditCard"]}
```

Comment:
isn't possible to use the keys here as the formik type from the `BillingInfoFormContext`?
I'm not sure but I guess we can override the `FormikProps<Pick<BillingInfoFormValues, key in FormKeys>>` so in the render method below you won't need to specify the type and import any extra type.
```jsx
{formik => ()}
```

Reply:
I get the concept but I guess `addressWithPhone` would be a problem here bc of what I mentioned [here](https://github.com/artsy/force/pull/9115#discussion_r777584355)?

Reply:
I guess you can just ignore my request, it's going too much into this mapping thing, but has no relation with your previous comment though.